components/dependency_injection/factories.rst

```diff
@@ -68,6 +68,13 @@ class:
 
         $container->setDefinition('newsletter_manager', $definition);
 
+.. note::
+
+    When creating a service through a factory, the value chosen for the ``class``
+    option has no actual effect since the final class only depends on the object
```

Comment:
"has no effect on the resulted service, since the actual class name only depends on the object returned by the factory."